src/main/java/de/retest/web/WebData.java

```diff
@@ -0,0 +1,87 @@
+package de.retest.web;
+
+import java.awt.Rectangle;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WebData {
+
+	private static final Logger logger = LoggerFactory.getLogger( WebData.class );
+
+	private final Map<String, Object> wrappedData;
+
+	public WebData( final Map<String, Object> wrappedData ) {
+		this.wrappedData = wrappedData;
+	}
+
+	/**
+	 * @throws IllegalStateException
+	 *             if the value for the given key is not a String.
+	 */
+	public String getAsString( final String key ) {
+		final Object result = wrappedData.get( key );
+		if ( result == null ) {
+			return null;
+		}
+		if ( result instanceof String ) {
+			return normalize( (String) result );
+		}
+		throw new IllegalStateException(
+				"Attribute value for " + key + " was of " + result.getClass() + ", not of type String." );
+	}
+
+	protected static String normalize( final String value ) {
+		if ( value == null ) {
+			return value;
+		}
+		String result = value;
+		if ( result.startsWith( "\"" ) && result.endsWith( "\"" ) ) {
+			result = result.substring( 1, result.length() - 1 );
+		}
+		return result.trim();
+	}
+
+	public Rectangle getOutline() {
+		if ( wrappedData.get( AttributesConfig.X ) == null || wrappedData.get( AttributesConfig.Y ) == null //
+				|| wrappedData.get( AttributesConfig.WIDTH ) == null
+				|| wrappedData.get( AttributesConfig.HEIGHT ) == null ) {
+			return null;
+		}
+		try {
+			final int x = getAsInt( AttributesConfig.X );
+			final int y = getAsInt( AttributesConfig.Y );
+			final int width = getAsInt( AttributesConfig.WIDTH );
+			final int height = getAsInt( AttributesConfig.HEIGHT );
+			return new Rectangle( x, y, width, height );
+		} catch ( final Exception e ) {
+			logger.error( "Exception retrieving outline: ", e );
+		}
+		return null;
+	}
+
+	/**
+	 * @throws IllegalStateException
+	 *             if the value for the given key cannot be converted to integer.
+	 */
+	public int getAsInt( final String key ) {
+		final Object value = wrappedData.get( key );
+		try {
+			if ( value instanceof Integer ) {
+				return (Integer) value;
+			}
+			if ( value instanceof String ) {
+				return Integer.parseInt( (String) value );
+			}
+			if ( value instanceof Double ) {
+				return Math.toIntExact( Math.round( (Double) value ) );
+			}
+			if ( value instanceof Long ) {
+				return Math.toIntExact( Math.round( (Long) value ) );
+			}
+		} catch ( final Exception e ) {}
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Either remove or fill this block of code. ([squid:S00108](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS00108))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=42&issues=AWT5lbh6J00hBq3sh_PK&open=AWT5lbh6J00hBq3sh_PK)

[ISSUE_KEY]: AWT5lbh6J00hBq3sh_PK